src/lib/helpers.ts

```diff
@@ -102,20 +96,72 @@ export function safeParseQueryString<T>(params: SafeParseQueryStringParams<T>):
 }
 
 /**
+ * This is an alternative for query-string parse method
  * Parse a query string into an object
  * @param query the query string
  */
-export const parseQueryString = queryString.parse;
+type ParsedQuery = Record<string, string | string[]>;
+export function parseQueryString(queryString: string): ParsedQuery {
+  const params = new URLSearchParams(queryString);
+  const parsedQuery: ParsedQuery = {};
+  for (const [key, value] of params.entries()) {
+    const normalizedKey = key.replace('[]', '');
+    const currentValue = parsedQuery[normalizedKey];
+    if (Array.isArray(currentValue)) {
+      parsedQuery[normalizedKey] = [...currentValue, value];
+    } else if (currentValue !== undefined) {
+      parsedQuery[normalizedKey] = [currentValue, value];
+    } else {
+      parsedQuery[normalizedKey] = key.includes('[]') ? [value] : value;
+    }
+  }
+
+  return parsedQuery;
+}
 
 /**
+ * @returns A function that takes a key and returns a function that encodes the key-value pair.
+ */
+function encoderForArrayFormat() {
```

Comment:
What are the things here being joined with the `join('')`?

Reply:
This function is primarily utilised for conducting searches with multiple values, such as on the work summary page, where users can search by multiple status values. Here, it concatenates the encoded key and value pair with the sign ([]=) and then encodes the resulting string. Finally, the entire array of key-value pairs is concatenated using the join method, which results in a single string containing all the encoded key-value pairs separated by the ampersand delimiter.

Reply:
@sabrine33 OK. Where you have `[encodeURIComponent(key), '[]'].join('')`, that is another way of writing `encodeURIComponent(key) + '[]'` I think.

Reply:
@sabrine33 OK. Where you have `[encodeURIComponent(key), '[]'].join('')`, that is another way of writing `encodeURIComponent(key) + '[]'` I think.

Reply:
yes, basically we are joining the elements of array using the specified join ('' here )

Reply:
@sabrine33 What prompted the removal of the query-string module?